File: clojure/src/project_euler/sol/p0099.clj
(ns project-euler.sol.p0099
  (:require
   [clojure.math :as math]
   [clojure.string :as str]
   [project-euler.lib.util :as util]))

(defn- parse-data
  [data]
  (->> (map #(str/split % #",") data)
       (map #(map parse-long %))))

(defn solve
  ([]
   (solve "0099_base_exp.txt"))
  ([fname]
   (let [xf (comp (map (fn [[b e]] (* e (math/log10 b))))
                  (map-indexed vector))]
     (->> (apply max-key second (into [] xf (parse-data (util/read-data fname))))
          (first)
          (inc)))))

File: julia/Euler/src/bin/p0099.jl

# project euler: problem 99

module Prob0099

function solve_0099(fname::String = "0099_base_exp.txt")
    answer = 0
    max_value = 0
    for (idx, x) in pairs([parse(Int, x) for x in split(l, ",")] for l in readlines(joinpath((@__DIR__), "../../assets", fname)))
        if (tmp = x[2] * log10(x[1])) > max_value
            answer = idx
            max_value = tmp
        end
    end
    answer
end

end #module

using .Prob0099: solve_0099
export solve_0099
